src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -4624,6 +4654,18 @@ theorem getLsbD_intMax (w : Nat) : (intMax w).getLsbD i = decide (i + 1 < w) :=
   · simp [h]
   · rw [Nat.sub_add_cancel (Nat.two_pow_pos (w - 1)), Nat.two_pow_pred_mod_two_pow (by omega)]
 
+@[simp]
+theorem toInt_intMax : (BitVec.intMax w).toInt = 2 ^ (w - 1) - 1 := by
+  have : intMax w = BitVec.twoPow w (w - 1) - 1#w := by simp [intMax]
+  rw [this]
+  rcases w with _|_|w
+  · decide
+  · decide
+  · have : 1 < 2 ^ (w + 1 + 1) := Nat.one_lt_two_pow (by omega)
+    rw_mod_cast [BitVec.twoPow, BitVec.toInt_sub, BitVec.toInt_shiftLeft, BitVec.toNat_ofNat,
+      Int.bmod_sub_bmod_congr, toInt_one_of_lt (by omega), Nat.shiftLeft_eq,
+      Nat.mod_eq_of_lt (by omega), Int.bmod_eq_of_le_of_lt (by omega) (by rw [← Nat.two_pow_pred_add_two_pow_pred (w := w + 1 + 1) (by omega)]; omega)]
```

Comment:
```suggestion
    rw_mod_cast [BitVec.twoPow, BitVec.toInt_sub, BitVec.toInt_shiftLeft, BitVec.toNat_ofNat,
      Int.bmod_sub_bmod_congr, toInt_one_of_lt (by omega), Nat.shiftLeft_eq,
      Nat.mod_eq_of_lt (by omega),
      Int.bmod_eq_of_le_of_lt (by omega)
          (by rw [← Nat.two_pow_pred_add_two_pow_pred (w := w + 1 + 1) (by omega)]; omega)]
```

To be honest, I wonder if there are lemmas to be extracted here, especially the somewhat heroic rewrite of 

```lean
Int.bmod_eq_of_le_of_lt (by omega) (by rw [← Nat.two_pow_pred_add_two_pow_pred (w := w + 1 + 1) (by omega)
```

All good if there is not, but would be nice if there is.